activerecord/lib/rails/generators/active_record/model/model_generator.rb

```diff
@@ -48,7 +48,7 @@ def generate_application_record
 
         # Used by the migration template to determine the parent name of the model
         def parent_class_name
-          options[:parent] || determine_default_parent_class
+          options[:parent] || 'ApplicationRecord'
```

Comment:
Do we have a general rule of thumb for freezing strings?

This is run once in a generator, that will exit in a reasonable finite time. I don't see how much allocations or performance can we squeeze out of this single run.